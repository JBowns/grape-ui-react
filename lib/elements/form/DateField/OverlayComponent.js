"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.OverlayComponent = OverlayComponent;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require("styled-components");

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _styledSystem = require("styled-system");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var StyledOverlayComponent = (0, _styledComponents2["default"])('div')(_templateObject(), _styledSystem.position);

function OverlayComponent(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ["children"]);

  return _react2["default"].createElement(StyledOverlayComponent, {
    className: "DayPickerInput-OverlayWrapper"
  }, _react2["default"].createElement("div", {
    className: "DayPickerInput-OverlayWrapper--Offset"
  }, children));
}

OverlayComponent.propTypes = {
  children: _propTypes2["default"].node.isRequired,
  position: _propTypes2["default"].string
};
OverlayComponent.defaultProps = {
  position: 'absolute'
};