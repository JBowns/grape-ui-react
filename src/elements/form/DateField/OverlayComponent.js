import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { position } from 'styled-system';

const StyledOverlayComponent = styled('div')`
  ${position}
`;

export function OverlayComponent({ children, ...props }) {
  return (
    <StyledOverlayComponent className="DayPickerInput-OverlayWrapper">
      <div className="DayPickerInput-OverlayWrapper--Offset">
        {children}
      </div>
    </StyledOverlayComponent>
  );
}

OverlayComponent.propTypes = {
  children: PropTypes.node.isRequired,
  position: PropTypes.string,
};

OverlayComponent.defaultProps = {
  position: 'absolute',
};
