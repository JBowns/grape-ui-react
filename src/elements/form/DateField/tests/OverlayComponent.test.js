import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { OverlayComponent } from '../OverlayComponent';
import 'jest-styled-components';

const emptyTheme = {};

const assertReactElement = element => {
  const component = renderer.create(element);
  return component.toJSON();
};
describe('Overlay Component base', () => {
  it('should return an Overlay Component object', () => {
    const element = <ThemeProvider theme={emptyTheme}><OverlayComponent>I am an overlay component.</OverlayComponent></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return an Overlay Component with position set to relative', () => {
    const element = <ThemeProvider theme={emptyTheme}><OverlayComponent position="relative">I am relative.</OverlayComponent></ThemeProvider>;
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});
