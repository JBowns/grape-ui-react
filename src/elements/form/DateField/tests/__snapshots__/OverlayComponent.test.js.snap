// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Component base should return an Overlay Component object 1`] = `
<div
  className="DayPickerInput-OverlayWrapper"
>
  <div
    className="DayPickerInput-OverlayWrapper--Offset"
  >
    I am an overlay component.
  </div>
</div>
`;

exports[`Overlay Component base should return an Overlay Component with position set to relative 1`] = `
<div
  className="DayPickerInput-OverlayWrapper"
>
  <div
    className="DayPickerInput-OverlayWrapper--Offset"
  >
    I am relative.
  </div>
</div>
`;
